File: app/src/main/java/com/robotnec/budget/core/navigator/Navigator.java
package com.robotnec.budget.core.navigator;

import android.app.Activity;

import com.robotnec.budget.core.domain.Category;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface Navigator {
    void openAddTransactionScreen(NavigationBundle navigationBundle);

    void openAddCategoryScreen(NavigationBundle navigationBundle);

    void openAddAccountScreen(NavigationBundle navigationBundle);

    void openCategory(Activity activity, Category category);
}

File: app/src/main/java/com/robotnec/budget/core/navigator/Navigator.kt
package com.robotnec.budget.core.navigator

import android.app.Activity
import androidx.navigation.NavController

import com.robotnec.budget.core.domain.Category

/**
 * @author zak zak@swingpulse.com>
 */
interface Navigator {
    fun openAddTransactionScreen(navController: NavController)

    fun openAddCategoryScreen(navigationBundle: NavigationBundle<*>)

    fun openAddAccountScreen(navigationBundle: NavigationBundle<*>)

    fun openCategory(activity: Activity, category: Category)
}
